### PR TITLE
generate validation rules for int & float fields

### PR DESCRIPTION
**Pull request by narcoticfresh**
before you could POST 'true' and 'false' (booleans) and it would save it as int's (0 and 1). this is not a sane behavior.. add tests for module to test this..

### CODE REVIEW

File: src/Graviton/CoreBundle/Tests/Controller/ModuleControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+        $client->request('GET', '/core/module?q='.urlencode('eq(key,investment)'));
+        $response = $client->getResponse();
+        $results = $client->getResults();
+
+        $this->assertResponseContentType(self::COLLECTION_TYPE, $response);
+        $this->assertEquals('investment', $results[0]->key);
+        $this->assertEquals(1, count($results));
+
+        // get entry by id
+        $moduleId = $results[0]->id;
```
Comment (lapistano) [inline src/Graviton/CoreBundle/Tests/Controller/ModuleControllerTest.php]: I would start a separate Test here. 
So assert that the $results[0]->id is the as it is expected and use it as requirement for a next test.

File: src/Graviton/CoreBundle/Tests/Controller/ModuleControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
```
Comment (lapistano) [inline src/Graviton/CoreBundle/Tests/Controller/ModuleControllerTest.php]: same as above

File: src/Graviton/CoreBundle/Tests/Controller/ModuleControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+     */
+    public function testDeleteModule()
+    {
+        // get id first..
+        $client = static::createRestClient();
+        $client->request('GET', '/core/module?q='.urlencode('eq(key,investment)'));
+        $results = $client->getResults();
+
+        // get entry by id
+        $moduleId = $results[0]->id;
```
Comment (lapistano) [inline src/Graviton/CoreBundle/Tests/Controller/ModuleControllerTest.php]: same as above.
Separating these tests will probably result in reducing code duplication as well.

Comment (narcoticfresh) [inline src/Graviton/CoreBundle/Tests/Controller/ModuleControllerTest.php]: Well, that is my problem - I don't know the record IDs..

`/core/module` is a service where we decided to not fix the ID - the user should lookup by the fixed field `key` (what we called "string id") instead. So I need to do that also in the test.

That's why I need to do this that way. I just wanted to do it once - but each time `setUp()` is called, i get potentially new IDs for the records, so I need to re-query them in every test..

I did the whole test basically only for the `testPostInvalidInteger` test - to test the thing i introduced here.. Also, this is all redundant to `AppControllerTest` anyway..

Maybe we should only test `testPostInvalidInteger` here and then test all the things the generator does (as this service is also generated) in a central `ShowcaseControllerTest` that needs to be done in a separate PR.

Do you know how to make it nicer under these circumstance (not knowing the ID & need to re-fetch every time)?

**Comment by narcoticfresh** [comment]
I can obviously do the "key-to-ID" lookup in a separate function, that would reduce code duplication - but I cannot make an `assert` on the ID itself.. And I think it's that what you imply..

File: src/Graviton/CoreBundle/Tests/Controller/ModuleControllerTest.php (hunk 3 above)
Comment (lapistano) [inline src/Graviton/CoreBundle/Tests/Controller/ModuleControllerTest.php]: ok, did not know that there are no fixed ids. I leave the tests here as well, even if they are verifying stuff already tested elsewhere. 
So under this circumstances I currently do not see another way as well .

Comment (hairmare) [inline src/Graviton/CoreBundle/Tests/Controller/ModuleControllerTest.php]: We should actually ditch stringId if what @narcoticfresh desecribes is it's sole purpose here. There is now reason not to use id and fix upserting and POSTing those.

**Pull request merged**